CHANGELOG.md

```diff
@@ -5,12 +5,114 @@ Documentation of all notable changes to the **Fizzy** project.
+- There is now `lib/fizzy/cxx` directory with substitute implementations of C++20 standard library features, which Fizzy
+  uses when compiled with pre-C++20 standard (default). It contains now `span` and `bit_cast` utilities.
+  [#535](https://github.com/wasmx/fizzy/pull/535) [#541](https://github.com/wasmx/fizzy/pull/541)
+
+### Changed
+
+- Optimization in `execute` to do allocate memory only once for all of the arguments, locals and data stack.
+  [#382](https://github.com/wasmx/fizzy/pull/382)
+- Trap handling in `execute` was simplified. [#425](https://github.com/wasmx/fizzy/pull/425)
+- `fizzy-bench` now uses WABT version 1.0.19. [#443](https://github.com/wasmx/fizzy/pull/443)
+- New simpler implementation of `f32.nearest` and `f64.nearest`. [#504](https://github.com/wasmx/fizzy/pull/504)
+- `execute.hpp` now includes declarations only for `execute` functions. Other functions of C++ API (`instantiate`,
+  `resolve_imported_functions`, exports access) were moved to `instantiate.hpp`.
+  [#518](https://github.com/wasmx/fizzy/pull/518)
+- Compiler warning settings now enable more strict checks. [#508](https://github.com/wasmx/fizzy/pull/508)
+- Parsing module with incorrect memory index now throws `validation_error` instead of `parser_error`.
+  [#546](https://github.com/wasmx/fizzy/pull/546)
+- Remove some less useful benchmarking cases. [#555](https://github.com/wasmx/fizzy/pull/555)
+- CI improvements. [#465](https://github.com/wasmx/fizzy/pull/465) [#516](https://github.com/wasmx/fizzy/pull/516)
+  [#519](https://github.com/wasmx/fizzy/pull/519) [#520](https://github.com/wasmx/fizzy/pull/520)
+  [#523](https://github.com/wasmx/fizzy/pull/523) [#532](https://github.com/wasmx/fizzy/pull/532)
+  [#549](https://github.com/wasmx/fizzy/pull/549)
+- Code and test cleanups and refactoring. [#495](https://github.com/wasmx/fizzy/pull/495)
+  [#505](https://github.com/wasmx/fizzy/pull/505) [#521](https://github.com/wasmx/fizzy/pull/521)
+  [#522](https://github.com/wasmx/fizzy/pull/522)
+
+### Fixed
+
+- Incorrect results in some floating point operations with NaNs in GCC builds without optimization.
+  [#513](https://github.com/wasmx/fizzy/pull/513)
+- Potential undefined behaviour in pointer arithmetic in parser. [#539](https://github.com/wasmx/fizzy/pull/539)
+- Ensure (in debug mode) that `execute()` receives the correct number of inputs and returns the correct number of output.
+  [#547](https://github.com/wasmx/fizzy/pull/547)
+- Incorrect failure logging in `fizzy-spectests` tool. [#548](https://github.com/wasmx/fizzy/pull/548)
+
 ## [0.4.0] — 2020-09-01
 
 This release introduces complete floating-point support.
 
 With that in place, Fizzy passes all(*) of the [official test suite (spectest 1.0)](https://github.com/wasmx/wasm-spec/tree/w3c-1.0-jsontests-20200813/test/core/json):
-  - 18896 of 18899 binary parser and execution tests,
+  - 17910 of 17913 binary parser and execution tests,
```

Comment:
This number actually included both parser/execution and validation. Separated it given we show validation separately.